tests/e2e/framework/kubernetes.go

```diff
@@ -172,47 +158,82 @@ func (k *KubeInfo) deployAddons() error {
-func (k *KubeInfo) deployIstioCore(module string) error {
-	yamlFile := filepath.Join(k.TmpDir, "yaml", fmt.Sprintf("istio-%s.yaml", module))
-	if err := k.generateIstioCore(yamlFile, module); err != nil {
+func (k *KubeInfo) deployIstio() error {
+	istioYaml := nonAuthInstallFile
+	if *authEnable {
+		istioYaml = authInstallFile
+	}
+	baseIstioYaml := util.GetResourcePath(filepath.Join(istioInstallDir, istioYaml))
+	testIstioYaml := filepath.Join(k.TmpDir, "yaml", istioYaml)
+
+	if *rbacEnable {
+		if *rbacfile == "" {
+			return errors.New("no rbac file is specified")
+		}
+		baseRbacYaml := util.GetResourcePath(*rbacfile)
+		testRbacYaml := filepath.Join(k.TmpDir, "yaml", filepath.Base(*rbacfile))
+		if err := k.generateRbac(baseRbacYaml, testRbacYaml); err != nil {
+			glog.Errorf("Generating rbac yaml failed")
+		}
+		if err := util.KubeApply(k.Namespace, testRbacYaml); err != nil {
+			glog.Errorf("Rbac deployment failed")
+			return err
+		}
+	}
+
+	if err := k.generateIstio(baseIstioYaml, testIstioYaml); err != nil {
+		glog.Errorf("Generating yaml %s failed", testIstioYaml)
 		return err
 	}
-	if err := util.KubeApply(k.Namespace, yamlFile); err != nil {
-		glog.Errorf("Kubectl apply %s failed", yamlFile)
+	if err := util.KubeApply(k.Namespace, testIstioYaml); err != nil {
+		glog.Errorf("Istio core %s deployment failed", testIstioYaml)
 		return err
 	}
 
 	return nil
 }
 
-func (k *KubeInfo) generateIstioCore(dst, module string) error {
-	src := util.GetResourcePath(filepath.Join(istioInstallDir, fmt.Sprintf("istio-%s.yaml", module)))
+func (k *KubeInfo) generateRbac(src, dst string) error {
 	content, err := ioutil.ReadFile(src)
 	if err != nil {
 		glog.Errorf("Cannot read original yaml file %s", src)
 		return err
 	}
-	var hubMacro, tagMacro string
-	var hubValue, tagValue []byte
-	switch module {
-	case "pilot":
-		hubMacro, tagMacro = `{PILOT_HUB}`, `{PILOT_TAG}`
-		hubValue, tagValue = []byte(*pilotHub), []byte(*pilotTag)
-	case "mixer":
-		hubMacro, tagMacro = `{MIXER_HUB}`, `{MIXER_TAG}`
-		hubValue, tagValue = []byte(*mixerHub), []byte(*mixerTag)
-	case "ingress":
-		hubMacro, tagMacro = `{PROXY_HUB}`, `{PROXY_TAG}`
-		hubValue, tagValue = []byte(*pilotHub), []byte(*pilotTag)
-	}
-	r := regexp.MustCompile(hubMacro)
-	content = r.ReplaceAllLiteral(content, hubValue)
-	r = regexp.MustCompile(tagMacro)
-	content = r.ReplaceAllLiteral(content, tagValue)
+	namespace := []byte(fmt.Sprintf("namespace: %s", k.Namespace))
+	r := regexp.MustCompile("namespace: default")
+	content = r.ReplaceAllLiteral(content, namespace)
+	err = ioutil.WriteFile(dst, content, 0600)
+	if err != nil {
+		glog.Errorf("Cannot write into generate rbac file %s", dst)
+	}
+	return err
+}
+
+func (k *KubeInfo) generateIstio(src, dst string) error {
+	content, err := ioutil.ReadFile(src)
+	if err != nil {
+		glog.Errorf("Cannot read original yaml file %s", src)
+		return err
+	}
+
+	if *mixerHub != "" || *mixerTag != "" {
```

Comment:
I was trying to match the situation that only hub or tag is changed, but now I realize, you are right, we don't need to consider that situation. So change to &&